### PR TITLE
Support retries in RequestHandler

### LINKED ISSUE #946: Add option to retry failed requests
We write our tests using webdriverio. But unfortunately our selenium grid is not stable enough or we experience network problems. The problem is that webdriverio can't get SessionId for other requests and mocha tests fail after timeout.

Currently `request` library is used to send requests to SG, but it doesn't support retries (or does it?) and it would be very helpful in our case. Would you mind if I make a PR which adds this option (retries for example) and also switches to `got`, `asker` etc?

### PR DESCRIPTION
**Pull request by 1999**
I added 3 retries into RequestHandler which fixes #946
Unfortunately I can't run tests on my mac - selenium-standalone seems to be broken.
Probably constants are a bad design here - should've I used them as an option to runner?

### CODE REVIEW

File: lib/utils/RequestHandler.js [hunk 1]
```diff
@@ -12,8 +12,57 @@ var Q = require('q'),
     ErrorHandler = require('./ErrorHandler'),
     packageDotJson  = require('../../package.json');
 
+var REQUEST_TIMEOUT = 10000;
+var REQUEST_RETRY_COUNT = 3;
```
Comment (christian-bromann) [inline lib/utils/RequestHandler.js]: Can you make this user configurable?

**Comment by christian-bromann** [comment]
One comment otherwise LGTM!

File: lib/utils/RequestHandler.js (hunk 1 above)
Comment (1999) [inline lib/utils/RequestHandler.js]: I'll take a deeper look on how I can do this. Need 4 more days, sorry :(

**Comment by 1999** [comment]
Didn't understand about one comment :(

**Comment by 1999** [comment]
Just added configurable options and updated PR. Take a look please.

**Comment by christian-bromann** [comment]
Could you add a functional test too? You could use the nock library to block a random request twice and get it resolve the 3rd time.

**Comment by 1999** [comment]
Just tried nock for the first time, seems like everything is working as expected (?)

**Comment by 1999** [comment]
:eyes:

**Comment by christian-bromann** [comment]
Looking good, thanks!

**Comment by christian-bromann** [comment]
@1999 your PR seems to break the build? Can you check why?

**Comment by christian-bromann** [comment]
I am afraid I can't port this to the v4.0 branch. It caused failures with Sauce Connect.

**Comment by 1999** [comment]
Can I try to port this myself or it's not possible at all?

**Comment by christian-bromann** [comment]
Sure you can create a PR for the v4.0 branch.

**Comment by 1999** [comment]
okay, I'll have a closer look at this branch. When do you plan to release it btw?

**Pull request merged**